cfme/fixtures/cli.py

```diff
@@ -0,0 +1,87 @@
+from utils.version import get_stream
+from cfme.test_framework.sprout.client import SproutClient
+from utils.appliance import current_appliance
+from utils.conf import cfme_data, credentials
+from utils.log import logger
+import pytest
+from wait_for import wait_for
+from cfme.test_framework.sprout.client import SproutException
+
+
+@pytest.yield_fixture(scope="function")
+def dedicated_db_appliance(app_creds):
+    sp = SproutClient.from_config()
+    version = current_appliance.version.vstring
+    stream = get_stream(current_appliance.version)
+    apps, pool_id = sp.provision_appliances(
+        count=1, preconfigured=False, version=version, stream=stream)
+    pwd = app_creds['password']
+    client = apps[0].ssh_client
+    channel = client.invoke_shell()
+    stdin = channel.makefile('wb')
+    stdin.write("ap \n 8 \n 1 \n 1 \n 1 \n y \n {} \n {} \n \n".format(pwd, pwd))
+    wait_for(apps[0].is_dedicated_db_active)
+
+    yield apps[0]
+
+    sp.destroy_pool(pool_id)
+
+
+@pytest.yield_fixture(scope="function")
+def fqdn_appliance():
+    sp = SproutClient.from_config()
+    available_providers = set(sp.call_method('available_providers'))
+    required_providers = set(cfme_data['fqdn_providers'])
+    usable_providers = available_providers & required_providers
+    version = current_appliance.version.vstring
+    stream = get_stream(current_appliance.version)
+
+    for provider in usable_providers:
```

Comment:
**REQUIRED:** Some comment around this would be nice, it looks at first glance as though you are requesting multiple appliances because of the loop.